src/components/modebar/manage.js

```diff
@@ -170,6 +179,26 @@ function getButtonGroups(gd) {
         dragModeGroup.push('select2d', 'lasso2d');
     }
 
+    // accept pre-defined buttons as string
+    if(Array.isArray(buttonsToAdd)) {
+        var newList = [];
+        for(var i = 0; i < buttonsToAdd.length; i++) {
+            var b = buttonsToAdd[i];
+            if(typeof b === 'string') {
+                if(DRAW_MODES.indexOf(b) !== -1) {
+                    if(
+                        // fullLayout._has('ternary') ||
+                        fullLayout._has('mapbox') ||
```

Comment:
🔪 ternary, and I didn't think we want this in mapbox either, do we? We don't support shapes on either of those subplot types, right?

Reply:
I removed `ternary` and added a comment regarding `mapbox` in https://github.com/plotly/plotly.js/pull/4775/commits/1f196994c9cfff253a76f5af4af9dd7b69956732.

Reply:
OK fair enough